src/ui/public/timefilter/timefilter.js

```diff
@@ -105,10 +106,21 @@ uiModules
       return this.calculateBounds(this.time);
     };
 
+    Timefilter.prototype.getForceNow = function () {
+      const query = querystring.parse(window.location.search.replace(/^\?/, '')).forceNow;
```

Comment:
I want this to be a normal querystring parameter outside of the AngularJS based hash routing, because I assumed that Canvas itself might not be using the hash based routing.

Reply:
Ah, fair enough. We *are* using hash routing though, and actually I think everything would need to. Though, I suppose an app *could* use search param routing...

So your intention is that you'd have something like `http://localhost:5601/acd/app/canvas?forceNow=2001-01-01T00:00:00Z#/workpad/workpad-1ba0f600-47f0-4306-a587-17cc95a74281`?

Reply:
Yup, that's the intent right now.

Reply:
What happens if an app needs the `forceNow` parameter to be part of the hash route? 

I think that's actually going to be the case for Canvas. Our router is going to assume that the entirety of the parameters it should be looking at come after the hash. I can work around it, but I wonder if the location of this setting can be left up to the app creating the URL somehow.

Reply:
Allowing the application to decide where to look for the parameter in the URL is going against the intent of this change entirely. The goal is to make the applications abide by a standard "interface" and then they'll work with Reporting, changing where the `forceNow` is for different applications is the exact thing that we're trying to avoid by introducing application specific logic.

With that being said, if we want to move the `forceNow` to the hash parameters, we can, as we can support this within Kibana. There's been discussion about moving to non hash-based routing, but we can readdress this when this happens

Reply:
Fair enough. I'll add an issue in Canvas to address search params that could be in the URL already and we can push forward that way.